plugins/uefi/fu-uefi-device.c

```diff
@@ -0,0 +1,194 @@
+		return "authentication signing error";
+	if (status == FU_UEFI_DEVICE_STATUS_ERROR_PWR_EVT_AC)
+		return "AC power required";
+	if (status == FU_UEFI_DEVICE_STATUS_ERROR_PWR_EVT_BATT)
+		return "battery level is too low";
+	return NULL;
+}
+
+static void
+fu_uefi_device_to_string (FuDevice *device, GString *str)
+{
+	FuUefiDevice *self = FU_UEFI_DEVICE (device);
+	g_string_append (str, "  FuUefiDevice:\n");
+	g_string_append_printf (str, "    kind:\t\t\t%s\n",
+				fu_uefi_device_kind_to_string (self->kind));
+	g_string_append_printf (str, "    fw_class:\t\t\t%s\n", self->fw_class);
+	g_string_append_printf (str, "    capsule_flags:\t\t%" G_GUINT32_FORMAT "\n",
+				self->capsule_flags);
+	g_string_append_printf (str, "    fw_version:\t\t\t%" G_GUINT32_FORMAT "\n",
+				self->fw_version);
+	g_string_append_printf (str, "    fw_version_lowest:\t\t%" G_GUINT32_FORMAT "\n",
+				self->fw_version_lowest);
+	g_string_append_printf (str, "    last_attempt_status:\t%s\n",
+				fu_uefi_device_status_to_string (self->last_attempt_status));
+	g_string_append_printf (str, "    last_attempt_version:\t%" G_GUINT32_FORMAT "\n",
+				self->last_attempt_version);
+}
+
+FuUefiDeviceKind
+fu_uefi_device_get_kind (FuUefiDevice *self)
+{
+	g_return_val_if_fail (FU_IS_UEFI_DEVICE (self), 0);
+	return self->kind;
+}
+
+guint32
+fu_uefi_device_get_version (FuUefiDevice *self)
+{
+	g_return_val_if_fail (FU_IS_UEFI_DEVICE (self), 0x0);
+	return self->fw_version;
+}
+
+guint32
+fu_uefi_device_get_version_lowest (FuUefiDevice *self)
+{
+	g_return_val_if_fail (FU_IS_UEFI_DEVICE (self), 0x0);
+	return self->fw_version_lowest;
+}
+
+guint32
+fu_uefi_device_get_version_error (FuUefiDevice *self)
+{
+	g_return_val_if_fail (FU_IS_UEFI_DEVICE (self), 0x0);
+	return self->last_attempt_version;
+}
+
+FuUefiDeviceStatus
+fu_uefi_device_get_status (FuUefiDevice *self)
+{
+	g_return_val_if_fail (FU_IS_UEFI_DEVICE (self), 0);
+	return self->last_attempt_status;
+}
+
+guint32
+fu_uefi_device_get_capsule_flags (FuUefiDevice *self)
+{
+	g_return_val_if_fail (FU_IS_UEFI_DEVICE (self), 0x0);
+	return self->capsule_flags;
+}
+
+const gchar *
+fu_uefi_device_get_guid (FuUefiDevice *self)
+{
+	g_return_val_if_fail (FU_IS_UEFI_DEVICE (self), NULL);
+	return self->fw_class;
+}
+
+static void
+fu_uefi_device_init (FuUefiDevice *self)
+{
+}
+
+static void
+fu_uefi_device_finalize (GObject *object)
+{
+	FuUefiDevice *self = FU_UEFI_DEVICE (object);
+
+	g_free (self->fw_class);
+
+	G_OBJECT_CLASS (fu_uefi_device_parent_class)->finalize (object);
+}
+
+static void
+fu_uefi_device_class_init (FuUefiDeviceClass *klass)
+{
+	GObjectClass *object_class = G_OBJECT_CLASS (klass);
+	FuDeviceClass *klass_device = FU_DEVICE_CLASS (klass);
+	object_class->finalize = fu_uefi_device_finalize;
+	klass_device->to_string = fu_uefi_device_to_string;
+}
+
+FuUefiDevice *
+fu_uefi_device_new_from_entry (const gchar *entry_path)
+{
+	FuUefiDevice *self;
+	guint64 hardware_instance = 0;	/* FIXME */
```

Comment:
I know this was in the original code, but what is this supposed to represent?  Is it time to drop it?

Reply:
I think so. willfix.